core/src/banking_stage/latest_unprocessed_votes.rs

```diff
@@ -0,0 +1,1036 @@
+use {
+    super::immutable_deserialized_packet::{DeserializedPacketError, ImmutableDeserializedPacket},
+    agave_feature_set as feature_set,
+    itertools::Itertools,
+    rand::{thread_rng, Rng},
+    solana_perf::packet::PacketRef,
+    solana_runtime::{bank::Bank, epoch_stakes::EpochStakes},
+    solana_sdk::{
+        account::from_account,
+        clock::{Slot, UnixTimestamp},
+        hash::Hash,
+        program_utils::limited_deserialize,
+        pubkey::Pubkey,
+        slot_hashes::SlotHashes,
+        sysvar,
+    },
+    solana_vote_program::vote_instruction::VoteInstruction,
+    std::{
+        cmp,
+        collections::HashMap,
+        ops::DerefMut,
+        sync::{
+            atomic::{AtomicBool, AtomicU64, AtomicUsize, Ordering},
+            Arc, RwLock,
+        },
+    },
+};
+
+#[derive(PartialEq, Eq, Debug, Copy, Clone)]
+pub enum VoteSource {
+    Gossip,
+    Tpu,
+}
+
+/// Holds deserialized vote messages as well as their source, and slot
+#[derive(Debug, Clone)]
+pub struct LatestValidatorVotePacket {
+    vote_source: VoteSource,
+    vote_pubkey: Pubkey,
+    vote: Option<Arc<ImmutableDeserializedPacket>>,
```

Comment:
why do we store an Option here? If we do not have a body of the packet why would we store it at all?

Reply:
sounds like someone is reviewing from the github UI and missing a bunch of context :P